### PR TITLE
HPCC-30405 Jtrace CNullSpan Implementation

### PR DESCRIPTION
**Pull request by rpastrana**
- Defines CNullSpan
- Adds cppunits
- Defaults activespans to CnullSpan
- Removes activespan nullptr checks

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@ghalliday I think this is what you asked for. I'm not sure about the behavior in the second commit.

**Review by ghalliday — COMMENTED** [review]
Looks good.  A few minor comments

**Comment by rpastrana** [comment]
@ghalliday please review.

**Review by ghalliday — COMMENTED** [review]
@rpastrana please squash (and remove change left in by mistake?)

File: helm/hpcc/values.yaml [hunk 1]
```diff
@@ -25,6 +25,10 @@ global:
   logging:
     detail: 80
 
+  #
```
[the remark this answers is not in the document]
Comment (rpastrana) [reply helm/hpcc/values.yaml:28]: @ghalliday yep, removing. But do we care to keep it as enabled in the default values file?

File: helm/hpcc/values.yaml [hunk 2]
```diff
@@ -25,6 +25,10 @@ global:
   logging:
     detail: 80
 
+  # tracing sets the default tracing information for all components. Can be overridden locally
```
Comment (rpastrana) [inline helm/hpcc/values.yaml:28]: @ghalliday Added the comment I initially had intended to expose here

File: helm/hpcc/values.yaml [hunk 3]
```diff
@@ -25,6 +25,10 @@ global:
...
+  tracing:
+    disable: false
```
Comment (ghalliday) [inline helm/hpcc/values.yaml:30]: Picky.  This option should be disabled for consistency for all the other uses.

**Review by ghalliday — APPROVED** [review]
@rpastrana I can merge as is, or you can include a change to rename the option.  The option should also be in the schema - with a description, because I am not quite sure of its effect.

File: system/jlib/jtrace.cpp [hunk 4]
```diff
@@ -814,7 +841,7 @@ class CTraceManager : implements ITraceManager, public CInterface
     Expected Configuration format:
     global:
         tracing:                            #optional - tracing enabled by default
-            disable: true                   #optional - disable OTel tracing
+            disabled: true                   #optional - disable OTel tracing
```
Comment (ghalliday) [inline system/jlib/jtrace.cpp:844]: trivial: extra space

**Review by ghalliday — APPROVED** [review]
@rpastrana please squash (and remove 1 space)

**Comment by rpastrana** [comment]
@ghalliday squashed

**Pull request merged**